Assets/Scripts/Environmental/UI/HeartSystem.cs

```diff
@@ -48,18 +54,54 @@ public void AssociateWith(Player player)
      * maximum number of heart containers that are to be displayed on screen, and also
      * updates the sprites for all of those containers according to the player's health.
      */ 
-    void OnPlayerHealthChanged()
+    private void OnPlayerHealthChanged()
     {
         maxPlayerHeartContainers = (player.MaxHealth + 1) / healthPerHeart;
         SetVisibleHeartContainers();
         FillHearts();
+
+        // Critical health
+        if (player.Health <= 2 && player.Health != 0 && player.Health < player.MaxHealth)
+        {
+            playerAtCriticalHealth = true;
+            StartCoroutine(BlinkCriticalHealth());
+        }
+        else
+        {
+            playerAtCriticalHealth = false;
+        }
+    }
+
+
+    /* Repeatedly flashes the lowermost heart container when the player is low on health.
+     */ 
+    private IEnumerator BlinkCriticalHealth()
+    {
+        Image firstHeart = heartImages[0];
+        Vector3 defaultHeartScale = heartImages[0].transform.localScale;
+
+        while (playerAtCriticalHealth)
+        {
+            if (firstHeart.transform.localScale == defaultHeartScale)
+            {
+                firstHeart.transform.localScale = Vector3.zero;
+            }
+            else
+            {
+                firstHeart.transform.localScale = defaultHeartScale;
+            }
+
+            yield return new WaitForSeconds(delayBetweenCriticalHealthFlashes);
+        }
+
+        firstHeart.transform.localScale = defaultHeartScale;
     }
 
 
     /* Loops through all heart containers the game supports and enables/disables them according
      * to the player's current max heart containers.
      */
-    void SetVisibleHeartContainers()
+    private void SetVisibleHeartContainers()
```

Comment:
I think the default access modifier for class methods/properties is private.

Reply:
I know, but I figured it doesn't hurt to be consistent.